File: src/FrontEnd/src/domain/game-state.ts
import { Cell } from './cell';
import { Player } from './player';

export type GameState = {
  cells: Cell[];
  players: Player[];
  gameId: GameId;
  gameStatus: GameStatus
};

export type GameId = string;
export type GameStatus =
  | 'NewGame'
  | 'MoveSuccess'
  | 'AiWin'
  | 'Player0Win'
  | 'Player1Win'
  | 'MoveSuccessNextPlayerSkipped'
  | 'MoveInvalid'
  | 'GameOver' | 'Draw';

File: src/Hive.Api/DTOs/GameState.cs
using System.Collections.Generic;
using Hive.Domain.Entities;

namespace Hive.Api.DTOs;

public sealed record GameState(
    string GameId,
    GameStatus GameStatus,
    IList<Player> Players,
    ISet<Cell> Cells,
    ICollection<HistoricalMove>? History = null
)
{
    public ICollection<HistoricalMove> History { get; } = History ?? new List<HistoricalMove>();
}